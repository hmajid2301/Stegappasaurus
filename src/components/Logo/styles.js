import { StyleSheet } from 'react-native';


const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  
  text: {
    fontFamily: 'RobotoThin',
    fontSize: 20,
  },

  logo: {
    height: 25,
    width: 25,
  },
});

export default styles;
