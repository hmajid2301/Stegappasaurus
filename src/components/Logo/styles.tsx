import { StyleSheet } from "react-native";

import { fonts } from "~/constants";

const styles = StyleSheet.create({
  logoContainer: {
    flexDirection: "row",
    flexWrap: "wrap"
  },

  text: {
    fontFamily: fonts.header,
    fontSize: 20
  },

  logo: {
    height: 25,
    width: 25
  }
});

export default styles;
