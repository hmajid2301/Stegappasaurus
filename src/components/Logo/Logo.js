import PropTypes from 'prop-types';
import React from 'react';
import { Image, Text, View } from 'react-native';

import styles from './styles';


const logo = require('../../assets/images/logo.png');


const Logo = ({ theme }) => (
  <View style={styles.container}>
    <Text style={[styles.text, { color: theme.color }]}>Steg</Text>
      <Image source={logo} style={styles.logo}/>
    <Text style={[styles.text, { color: theme.color }]}>ppasaurus</Text>
  </View>
);

Logo.propTypes = {
  theme: PropTypes.object.isRequired,
};

export default Logo;
