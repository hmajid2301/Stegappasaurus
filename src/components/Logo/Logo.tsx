import * as React from 'react';
import {Image, Text, View} from 'react-native';

import logoDark from '~/assets/images/logo-dark.png';
import logoLight from '~/assets/images/logo-light.png';
import {ThemeColors} from '~/constants/types';
import styles from './styles';

interface IProps {
  isDark: boolean;
  color: ThemeColors;
}

const Logo = ({color, isDark}: IProps) => (
  <View style={styles.logoContainer}>
    <Text style={[styles.text, {color}]}>Steg</Text>
    <Image source={isDark ? logoLight : logoDark} style={styles.logo} />
    <Text style={[styles.text, {color}]}>ppasaurus</Text>
  </View>
);

export default Logo;
