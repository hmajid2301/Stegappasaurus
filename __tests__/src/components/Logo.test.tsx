import {shallow} from 'enzyme';
import React from 'react';

import Logo from '~/components/Logo';

describe('Logo: Match snapshots', () => {
  test('1', () => {
    const component = shallow(<Logo color="#17212D" isDark={true} />);
    expect(component).toMatchSnapshot();
  });

  test('2', () => {
    const component = shallow(<Logo color="#FFF" isDark={false} />);
    expect(component).toMatchSnapshot();
  });
});
