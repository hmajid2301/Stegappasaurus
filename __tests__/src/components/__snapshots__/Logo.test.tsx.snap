// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo: Match snapshots 1 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto-Thin",
          "fontSize": 20,
        },
        Object {
          "color": "#17212D",
        },
      ]
    }
  >
    Steg
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/logo-light.png",
      }
    }
    style={
      Object {
        "height": 25,
        "width": 25,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto-Thin",
          "fontSize": 20,
        },
        Object {
          "color": "#17212D",
        },
      ]
    }
  >
    ppasaurus
  </Text>
</View>
`;

exports[`Logo: Match snapshots 2 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto-Thin",
          "fontSize": 20,
        },
        Object {
          "color": "#FFF",
        },
      ]
    }
  >
    Steg
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/logo-dark.png",
      }
    }
    style={
      Object {
        "height": 25,
        "width": 25,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto-Thin",
          "fontSize": 20,
        },
        Object {
          "color": "#FFF",
        },
      ]
    }
  >
    ppasaurus
  </Text>
</View>
`;
